independent-projects/tools/common/src/main/java/io/quarkus/devtools/project/extensions/ExtensionManager.java

```diff
@@ -0,0 +1,109 @@
+package io.quarkus.devtools.project.extensions;
+
+import io.quarkus.bootstrap.model.AppArtifactCoords;
+import io.quarkus.bootstrap.model.AppArtifactKey;
+import io.quarkus.dependencies.Extension;
+import io.quarkus.devtools.project.BuildTool;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * This interface defines a high level way of managing (read/write) extensions in any QuarkusProject
+ */
+public interface ExtensionManager {
+
+    /**
+     * @return the {@link BuildTool} of this extension manager
+     */
+    BuildTool getBuildTool();
+
+    /**
+     * Read the build file(s) to get the list of installed extensions in this Quarkus project.
+     *
+     * @return The list of {@link AppArtifactCoords} installed in the project build file(s).
+     * @throws IOException if a problem occurs while reading the project build file(s)
+     */
+    List<AppArtifactCoords> getInstalled() throws IOException;
+
+    /**
+     * Read build file(s) to check if an extension is installed in this Quarkus project.
+     *
+     * @param key the {@link AppArtifactKey} of the extension to check
+     * @return true if it's installed
+     * @throws IOException if a problem occurs while reading the project build file(s)
+     */
+    default boolean isInstalled(AppArtifactKey key) throws IOException {
+        return getInstalled().stream().anyMatch(i -> Objects.equals(i.getKey(), key));
+    }
+
+    /**
+     * Check that the Quarkus Platform bom is defined in the project.
+     * The Quarkus Platform Bom role is to define version to use for Quarkus extensions.
+     *
+     * @return true if it's defined, false else
+     * @throws IOException if a problem occurs while reading the project build file(s)
+     */
+    boolean hasQuarkusPlatformBom() throws IOException;
+
+    /**
+     * This is going to install/add all the specified extensions to the project build file(s).
+     *
+     * <pre>
+     *   - If the project Quarkus platform bom is not defined, an {@link IllegalStateException} will be thrown
+     *   - Extensions which are already installed will ALWAYS be skipped whatever the specified version
+     *   - The provided version will be used wasn't already installed
+     * </pre>
+     *
+     * @param coords the list of {@link AppArtifactCoords} for the extensions to install
+     * @return the {@link InstallResult}
+     * @throws IOException if a problem occurs while reading/writing the project build file(s)
+     */
+    InstallResult install(List<AppArtifactCoords> coords) throws IOException;
```

Comment:
I'd prefer to use `Collection` instead. That would prevent cases where the caller has a different Collection implementation and need to convert to List to call this method

Reply:
For this one I would really like to have a List since it's ordered and you may want to add them in a specific order..